### PR TITLE
feat: Add capability to display all types of warnings handled and the…

### PR DESCRIPTION
**Pull request by halcin**
…ir definitions

### CODE REVIEW

File: src/reporting/reporters/post-pipeline/console/warnings-description.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Import Third-party Dependencies
+import * as i18n from "@nodesecure/i18n";
+import * as jsxray from "@nodesecure/js-x-ray";
+import { table } from "table";
+
+// Import Internal Dependencies
+import { consolePrinter } from "../../../../../lib/console-printer/index.js";
+import { Nsci } from "../../../../configuration/standard/index.js";
+
+export async function printWarningsDescription(): Promise<void> {
```
Comment (fraxken) [inline src/reporting/reporters/post-pipeline/console/warnings-description.ts:10]: There nothing async (so no need)

File: src/reporting/reporters/post-pipeline/console/warnings-description.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  const tableData = [
+    [
+      consolePrinter.font.standard("Warning label").bold().message,
+      consolePrinter.font.standard("Warning description").bold().message
+    ]
+  ];
+  const warningDescriptions: any[] = [];
```
Comment (fraxken) [inline src/reporting/reporters/post-pipeline/console/warnings-description.ts:17]: type `string[]` ?

File: src/reporting/reporters/post-pipeline/console/warnings-description.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    warningDescriptions.push(
+      i18n.getTokenSync(jsxray.warnings[warningName].i18n)
+    );
+    warningLabels.push(warningName);
+  }
+  for (let i = 0; i < warningLabels.length; i++) {
+    tableData.push([
+      warningLabels[i],
+      consolePrinter.font.standard(warningDescriptions[i]).message
+    ]);
```
Comment (fraxken) [inline src/reporting/reporters/post-pipeline/console/warnings-description.ts:29]: you can push if in the previous loop iteration no? No need for a second one

File: src/reporting/reporters/post-pipeline/console/index.ts [hunk 4]
```diff
@@ -58,5 +59,6 @@ export const consoleReporter: Reporter<
     const endedAt = performance.now() - startedAt;
     printEndPipeline(endedAt);
     printPipelineOutcome(data, status, warnings as Nsci.Warnings);
+    printWarningsDescription();
```
Comment (antoine-coulon) [inline src/reporting/reporters/post-pipeline/console/index.ts:62]: As it was mentioned in the issue, I feel like printing that table should be done via another CLI command and not being part of the "post-pipeline" step as it pollutes the global outcome.

You could create a specific CLI command in `bin/cli.ts` and invoke it from there

Comment (antoine-coulon) [reply src/reporting/reporters/post-pipeline/console/index.ts]: This file has not been moved but this is good enough for now.

**Review by antoine-coulon — APPROVED** [review]

**Comment by antoine-coulon** [comment]
Thanks for contributing @halcin 🎉

**Pull request merged**